pkg/registry/authorization/selfsubjectaccessreview/rest_test.go

```diff
@@ -0,0 +1,140 @@
+package selfsubjectaccessreview
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	authorizationv1 "github.com/openshift/api/authorization/v1"
+	authorizationscope "github.com/openshift/apiserver-library-go/pkg/authorization/scope"
+
+	authorizationapi "k8s.io/kubernetes/pkg/apis/authorization"
+)
+
+type fakeAuthorizer struct {
+	attrs authorizer.Attributes
+}
+
+func (f *fakeAuthorizer) Authorize(ctx context.Context, attrs authorizer.Attributes) (authorizer.Decision, string, error) {
+	f.attrs = attrs
+	return authorizer.DecisionNoOpinion, "", nil
+}
+
+func TestCreate(t *testing.T) {
+	userNilExtra := &user.DefaultInfo{}
+
+	userNoExtra := &user.DefaultInfo{
+		Extra: make(map[string][]string),
+	}
+
+	userNoScopes := &user.DefaultInfo{
+		Extra: map[string][]string{
+			"extra": {"ex1", "ex2"},
+		},
+	}
+
+	userWithScopesNoCheckAccess := &user.DefaultInfo{
+		Extra: map[string][]string{
+			"extra": {"ex1", "ex2"},
+			authorizationv1.ScopesKey: {
+				authorizationscope.UserInfo,
+				authorizationscope.UserListAllProjects,
+			},
+		},
+	}
+
+	userWithScopesWithCheckAccess := &user.DefaultInfo{
+		Extra: map[string][]string{
+			"extra": {"ex1", "ex2"},
+			authorizationv1.ScopesKey: {
+				authorizationscope.UserAccessCheck,
+				authorizationscope.UserInfo,
+			},
+		},
+	}
+
+	userWithScopeUserFull := &user.DefaultInfo{
+		Extra: map[string][]string{
+			"extra": {"ex1", "ex2"},
+			authorizationv1.ScopesKey: {
+				authorizationscope.UserAccessCheck,
+				authorizationscope.UserInfo,
+				authorizationscope.UserFull,
+			},
+		},
+	}
+
+	testcases := map[string]struct {
+		user         user.Info
+		expectedUser user.Info
+	}{
```

Comment:
you may want to add a `role:` test here, too, to see that the behavior matches the expectation based on the return value of `scopesNeedUserFull()`

Reply:
Good idea; added.